proto/dictation_asr.proto

```diff
@@ -418,46 +506,110 @@ message StreamingRecognizeResponse {
     // until the server closes the gRPC connection. This event is only sent if
     // `single_utterance` was set to `true`, and is not used otherwise.
     END_OF_SINGLE_UTTERANCE = 1;
-    
+
+    // [**Extension by Techmo**]
+    // This event indicates that recognition has started and there was no speech detected
+    // for a certain period of time. The server will not process
+    // additional audio (although it may subsequently return additional results).
+    // The client should stop sending additional audio data, half-close the gRPC
+    // connection, and wait for any additional results until the server closes
+    // the gRPC connection.
+    // MRCPv2-related: [No-Input-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.6)
+    NO_INPUT_TIMEOUT = 2;
+
     // [**Extension by Techmo**]
-    // This event indicates that the server has detected no speech input timeout
-    // and expects no additional speech. Therefore, the server will not process
+    // This event indicates that recognition has started and there was no result
+    // for a certain period of time. The server will not process
     // additional audio (although it may subsequently return additional results).
     // The client should stop sending additional audio data, half-close the gRPC
     // connection, and wait for any additional results until the server closes
     // the gRPC connection.
-    NO_INPUT_TIMEOUT = 4;
+    // MRCPv2-related: [Recognition-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.7)
+    RECOGNITION_TIMEOUT = 3;
+
+    // [**Extension by Techmo**]
+    // [Speech-Complete-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.15)
+    // This event indicates that recognition has been finalized with a complete result
+    // after specified length of silence after user speech.
+    SPEECH_COMPLETE_TIMEOUT = 4;
+
+    // [**Extension by Techmo**]
+    // [Speech-Incomplete-Timeout](https://tools.ietf.org/html/rfc6787#section-9.4.16)
+    // This event indicates that recognition has been finalized with an incomplete result
+    // after specified length of silence after user speech.
+    SPEECH_INCOMPLETE_TIMEOUT = 5;
   }
 
-  // [*Output-only*] If set, returns a [google.rpc.Status][google.rpc.Status] message that
+  // [**Extension by Techmo**]
+  // Indicates the type of speech processing timeout. These are MRCPv2-related.
+//  enum SpeechProcessingTimeoutType {
```

Comment:
Why it's commented here?

Reply:
I did not clean up after my previous idea. I will remove it.